### ISSUE TITLE
SyntaxError: Unexpected token <(…) 

### ISSUE THREAD

**Issue by Znow** [KallynGowdy/angular2-firebase#8]
Hi

As per your documentation and example app, I've implemented the bootstrap stuff, which gives me:

![image](https://cloud.githubusercontent.com/assets/265719/13349531/e6017d16-dc7a-11e5-836f-00b6d190616c.png)

``` typescript
import {bootstrap}    from 'angular2/platform/browser';
import {AppComponent} from './app.component';
import {ROUTER_PROVIDERS} from 'angular2/router';
import {provide} from 'angular2/core';
import {FirebaseService} from 'firebase-angular2/core';
import {FirebasePostService} from './post/firebase-post.service';
import {PostService} from './post/post.service';

declare var Firebase;

bootstrap(AppComponent, [
  ROUTER_PROVIDERS,
  provide(PostService, {useFactory: () => new FirebasePostService(new FirebaseService(new Firebase('https://znow.firebaseio.com/')))})
]);
```

If I outcomment my "provide" in the bootstrap, I wont get the error.

Kind regards

**Comment by KallynGowdy** [comment]
Hi Znow,

This appears to be an error that is produced by `system.js` that occurs when parsing a file. More than likely, the syntax error occurs in the `./post/firebase-post.service` module, as the reference would not be emitted by the TypeScript compiler when the `provide(...)` statement is commented out, which means that `system.js` would not need to parse the related JS file.

Alternatively, if you think that the syntax error is occurring in the `firebase-angular2/core` module, I can look it over, but I am fairly confident that the syntax error does not occur in one of the `firebase-angular2` JavaScript files.

In addition, there are some StackOverflow posts that might help you out. One such example that appears to be related to your case is: [Angular 2 Quickstart: unexpected token <](http://stackoverflow.com/a/35418193/1832856)

If you find a solution, please comment with it and I will close this issue. Otherwise, comment with whatever additional evidence you can find for the problem. If I don't hear back in about 5 days, I will close this issue as resolved anyways.

**Comment by Znow** [comment]
Hi @KallynGowdy 

This is the content of the firepast-post.service.ts:

``` typescript
import {Injectable} from "angular2/core";
import {Observable} from "rxjs/Rx";
import {FirebaseService} from 'firebase-angular2/core';
import {PostService} from "./post.service";
import {POSTS} from './mock-posts';
import {Post} from "./post";

@Injectable()
export class FirebasePostService extends PostService {
  private service: FirebaseService;

  constructor(firebaseService: FirebaseService) {

    this.service = firebaseService.child('posts');
    // Required for derived class
    super();
  }

  getPosts() {
    var service = this.service;
    return service.value.map((posts) => {
        return posts.map((p, i) => {
            // TODO: Cleanup
            return {
                id: p.id,
                title: p.title,
                save: function () {
                    return service.child(i.toString()).setData({
                        id: this.id,
                        title: this.title
                    });
                }
            }
        })
    });
  }
}
```

Alot like your "firebase-hero.service.ts" in your example application.

The rest of my code can be found in my repo: [https://github.com/Znow/ZnowBlog/tree/master/app](url)

Couldn't find anything that could probably solve my issue from the link you mentioned.

I also tried cloning your demo application repo, and run it locally, but it wouldn't load up at all.

Kind regards

**Comment by Znow** [comment]
Hi again

Even if I just use the example from your firebase-angular2 repo:

``` typescript
import {bootstrap}    from 'angular2/platform/browser';
import {ROUTER_PROVIDERS} from 'angular2/router';
import {provide, Inject} from 'angular2/core';
import {AppComponent} from './app.component';
import {FirebaseServiceProvider, FirebaseService} from 'firebase-angular2/core';
import {FirebasePostService} from './post/firebase-post.service';
import {PostService} from './post/post.service';

declare var Firebase;

bootstrap(AppComponent, [
  ROUTER_PROVIDERS,
  /*PostService,*/
  provide(FirebaseService, {useFactory: () => new FirebaseService(new Firebase('https://znow.firebaseio.com/'))})
]);
```

then im getting the exact same error.

Kind regards

**Comment by KallynGowdy** [comment]
@Znow,

I looked at your repository, and I noticed that your [index.html](https://github.com/Znow/ZnowBlog/blob/master/index.html) contains some odd settings.

In particular, the default extension for all of your app scripts is set to `'ts'`, and not `'js'`.

``` js
System.config({
        transpiler: 'typescript',
        typescriptOptions: { emitDecoratorMetadata: true },
        packages: {
          'app': { defaultExtension: 'ts' } // Should change this to 'js'
        },
        map: {
         'marked': 'node_modules/marked/marked.min.js',
         'firebase-angular2': 'node_modules/firebase-angular2'
       }
      });
      System.import('app/main')
            .then(null, console.error.bind(console));
```

You should be compiling all of your TypeScript files into JavaScript files using the [typescript NPM package](https://www.npmjs.com/package/typescript). The reason this is a problem is because `system.js` will load your `ts` files, thinking that they are JavaScript, but they are in fact TypeScript. As a result, anywhere a `<` or a `>` is used correctly in TypeScript, but incorrectly in JavaScript, a SyntaxError exception occurs.

In addition, you can set up NPM scripts to build and run your application for you. If you look at the [`ts/package.json`](https://github.com/KallynGowdy/firebase-angular2-demo/blob/master/ts/package.json) file in my demo application, you will see commands that look like this:

``` json
"scripts": {
    "tsc": "tsc",
    "tsc:w": "tsc -w",
    "lite": "lite-server",
    "start": "concurrent \"npm run tsc:w\" \"npm run lite\" "
 },
```

These commands can be run from the console/terminal via `npm run {script-name}`, where you replace `{script-name}` with the name of the script that you want to run.

So, to sum it up:
1. Compile TypeScript files into JavaScript files. Use the [tsconfig.ts](https://github.com/KallynGowdy/firebase-angular2-demo/blob/master/ts/tsconfig.json) from the demo application as an example.
2. Change the `defaultExtension` setting from `'ts'` to `'js'`.
3. Use NPM scripts to automate the build and run cycle.

If this is the solution to your problem, please comment back and I will close the issue.

**Comment by Znow** [comment]
Hi @KallynGowdy 

Mistyped that setting, of course it should be set to 'js', but that doesn't change anything though, still getting the same error.

Thanks for the explanation.

Kind regards

**Comment by KallynGowdy** [comment]
@Znow,

Something else I noticed when I was building my demo application is that it helped to import `Rx.js` and `angular2.dev.js` via `system.js` before importing my app. (see [index.html](https://github.com/KallynGowdy/firebase-angular2-demo/blob/master/ts/index.html))  I would suggest that you attempt the same, and maybe it will work for you?

Also, I found a couple links that might be beneficial:
- [StackOverflow](http://stackoverflow.com/questions/35596861/uncaught-syntaxerror-unexpected-token-vm10517uncaught-referenceerror-system)
- [Google Search](https://www.google.com/search?q=system.js%20unexpected%20token)

Other than that, I don't know how to help. As such, I'm going to close the issue. Let us know if you find anything else that helps though.

Thanks!
